src/main.rs

```diff
@@ -80,12 +84,26 @@ fn main() -> Result<()> {
             log::warn!("Short delay: release any keys now!");
             std::thread::sleep(Duration::from_secs_f64(delay));
 
-            let device_info = deviceinfo::DeviceInfo::with_name(
-                &mapping_config.device_name,
-                mapping_config.phys.as_deref(),
-            )?;
+            let mut device_info;
+            let mut i = 0f64;
+            loop {
+                device_info = deviceinfo::DeviceInfo::with_name(
+                    &mapping_config.device_name,
+                    mapping_config.phys.as_deref(),
+                );
+                if !wait || device_info.is_ok() {
+                    break;
+                }
+                let msg = device_info.err();
+                if msg.is_some() {
+                    log::info!("{}", msg.unwrap());
+                }
```

Comment:
I don't think `info!` is the best log level, and I think logging on each iteration could be overwhelming/spammy, so how about something like this, keeping in mind that I think this bit should be factored out into a separate function:

```suggestion
                match deviceinfo::DeviceInfo::with_name(
                    &mapping_config.device_name,
                    mapping_config.phys.as_deref(),
                ) {
                   Ok(device) => return Ok(device),
                   Err(err) if !wait => return Err(err),
                   Err(err) => {
                      if is_first_iteration {
                         log::warn!("Did not find device {device_name} {phys}: {err:#}. Will wait for it to be plugged in");
                      }
                }
```